src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -783,6 +810,21 @@ private void GetMargin(double[] values, double[][] results, double sensitivity)
                 }
             }
 
+            // Adjust the expected value if original data range is non-negative or non-positive
+            private double AdjustExpectedValueBasedOnOriginalDataRange(double expectedValue)
+            {
+                if (_minimumOriginValue >= 0 && expectedValue < 0)
```

Comment:
If the minimum original value > 0 and you set the expected value = 0, couldn't this leave the expected value out of range? Is this adjusted elsewhere?

Reply:
I does't adjust the expected value that much, there are cases that the expected value curve is not smooth and looks odd. Currently for not anomaly points, we have logic that ensure it is close to the original value which can guarantee the expected value is not far away from the original data range.

---
In reply to: [505667560](https://github.com/dotnet/machinelearning/pull/5436#discussion_r505667560) [](ancestors = 505667560)

Reply:
Just curious does kensho have similar check in your service?

Reply:
kensho has similar logic to adjust the expected value of anomaly / non-anomaly points. As there is no dsat reported regarding data range so far, we don't implement the data range check.

---
In reply to: [506583086](https://github.com/dotnet/machinelearning/pull/5436#discussion_r506583086) [](ancestors = 506583086)